File: example/win32.c
#include <stdio.h>
#include <stdlib.h>

#include "sigwatch.h"

int main(void){

setbuf(stdout, NULL);
/* disable default stdout buffering for clarity */

/* do ^C once to get a signal. The second time stops the program. */
int status = watchsignal(2);
printf("watchsignal 2: %d\n", status);

for(int i=0; i<10; i++){
  c_sleep(500);
  printf("lastsig=%d\n", getlastsignal());
}

printf("C: stopped watching for signals\n");

return EXIT_SUCCESS;

}

File: example/win32.f90
program sigs

use, intrinsic :: iso_c_binding, only: C_INT

implicit none (type, external)

integer :: i, status

interface
integer(C_INT) function watchsignal(sig) bind(C)
import C_INT
integer(C_INT), intent(in), value :: sig
end function

integer(C_INT) function getlastsignal() bind(C)
import C_INT
end function

subroutine milli_sleep(millseconds) bind(C, name="c_sleep")
import C_INT
integer(C_INT), intent(in), value :: millseconds
end subroutine

end interface

!! do ^C once to get a signal. The second time stops the program.
status = watchsignal(2)
print '("watchsignal 2:",i2)', status


do i=1,10
  call milli_sleep(500)
  print '("lastsig=", i2)', getlastsignal()
enddo

print '(a)', "Fortran: stopped watching for signals"

end program
